mq/inmemory/mq.h

```diff
@@ -25,7 +25,7 @@ class MMQ final {
   typedef MESSAGE T_MESSAGE;
 
   // Type of the processor of the entries.
-  // It should expose one method, void OnMessage(const T_MESSAGE&, size_t number_of_dropped_messages_if_any);
+  // It should expose one method, void OnMessage(T_MESSAGE&, size_t number_of_dropped_messages_if_any);
```

Comment:
There is a longer comment in Sherlock.

Gist: To allow passing data between message-queue-s of polymorphic types by doing `queue2.emplace_back(unique_ptr_from_queue.release())`.

If immutable objects are passed, the job of cloning the instance of a possibly polymorphic type becomes the job of the client, which is not good.

Also, by design, the element is only added to the message queue once and only removed from the message queue once. Thus, non-immutable accessor makes more sense performance-wise.